### DISCUSSION TITLE
Can't start server with: [custom formatter threw an exception]

### DISCUSSION THREAD

**Discussion by vedantroy** [vercel/next.js#73662] [Help]
### Summary

I'm getting the following error when trying to launch my NextJS script:
```
 ⨯ Failed to start server
[custom formatter threw an exception]
```
This happens on my VPS, but on my dev machine. No idea what's going on. Thoughts?

**Comment by llPekoll** [comment]
I go this one to!
Bu it was during building time on the VPS `UBUNTU`,  but build was working on my machine `OSX`

**Reply by eduvhc** [reply]
Same happening here. Running Dockerfile in coolify

**Reply by eduvhc** [reply]
I resolved the issue by switching from the `bun` runtime to Node.js with `pnpm`, and everything is working perfectly now.

**Reply by VivekGupta137** [reply]
I was getting the issue when using bun, just installed the nodejs and now it's not showing this error. I'm still using bun

**Comment by rhkdgns95** [comment]
I have the same problem in bun environment.

However, it runs normally in local, but in docker environment, it occurs as above [custom formatter threw an exception].

Did you solve this issue?

**Comment by asrafilll** [comment]
I have the same problem aswell in my docker env + ubuntu server, not on my dev local (mac os)

**Comment by RobertBorg** [comment]
i ended up using a 3 stage Dockerfile as a workaround where the installer and final stage are running on bun image while the one that actually is running next build runs node.

```
FROM oven/bun:1.1.42-debian AS installer

# Set the working directory
WORKDIR /app

COPY transfer-protocol/ transfer-protocol/

# Copy package.json and bun.lockb
COPY frontend/package.json frontend/bun.lockb frontend/

WORKDIR /app/frontend
# Install dependencies
RUN bun install


FROM node:22.12.0-bookworm AS builder
COPY --from=installer /app /app
# Copy the rest of the application code
COPY frontend/ /app/frontend/
WORKDIR /app/frontend
RUN npm run build

FROM oven/bun:1.1.42-debian AS final

COPY --from=builder /app /app

WORKDIR /app/frontend
# Expose the port (default to 3000)
ARG PORT=3000
ENV PORT $PORT
EXPOSE $PORT

# Start the Next.js application
CMD ["bun", "run", "start"]
```

**Reply by sethgw** [reply]
same / similar problem

**Reply by kelola-tech** [reply]
Im having the same issue,

**Comment by FrancescoSaverioZuppichini** [comment]
same issue here

**Comment by akmaldira** [comment]
same issue here
